modules/client-cache/src/main/scala/higherkindness/mu/rpc/channel/cache/ClientCache.scala

```diff
@@ -34,22 +34,24 @@ object ClientCache {
 
   private val logger: Logger = getLogger
 
-  type HostPort = (String, Int)
-
-  def fromResource[Client[_[_]], F[_]](
-      getHostAndPort: F[HostPort],
-      createClient: HostPort => Resource[F, Client[F]],
+  def fromResource[Client[_[_]], F[_], H](
+      getHostAndPort: F[H],
+      createClient: H => Resource[F, Client[F]],
       tryToRemoveUnusedEvery: FiniteDuration,
       removeUnusedAfter: FiniteDuration
   )(
       implicit CE: ConcurrentEffect[F],
       cs: ContextShift[F],
       timer: Timer[F]): Stream[F, ClientCache[Client, F]] =
-    impl(getHostAndPort, createClient(_).allocated, tryToRemoveUnusedEvery, removeUnusedAfter)
-
-  def impl[Client[_[_]], F[_]](
-      getHostAndPort: F[HostPort],
-      createClient: HostPort => F[(Client[F], F[Unit])],
+    impl(
+      getHostAndPort,
+      (h: H) => createClient(h).allocated,
```

Comment:
I think we can keep the curried alternative
```scala
createClient(_).allocated
```

Reply:
I tried that but it can't infer type. It shows "missing parameter type" error :'(